File: python/7 kyu/359.py
# https://www.codewars.com/kata/628e3ee2e1daf90030239e8a/train/python

# Passed

def interlockable(a, b):
    a_binary = bin(a).split('b')[1]
    b_binary = bin(b).split('b')[1]
    
    result = True
    for group in zip(a_binary[::-1], b_binary[::-1]):
        if group == ('1', '1'):
            result = False
            break
            
    return result

output = interlockable(9, 4)
print(output)

File: nodejs/7 kyu/359.js
// https://www.codewars.com/kata/562f91ff6a8b77dfe900006e/train/javascript

// Passed

function movie(card, ticket, perc) {
  let systemACost = 0;
  let systemBCost = card;
  let systemBTicketCost = ticket * perc;
  let result = 0;

  while (systemACost <= Math.ceil(systemBCost)) {
    systemACost += ticket;
    systemBCost += systemBTicketCost;
    systemBTicketCost *= perc;
    result++;
  }

  return result;
}

const output = movie(500, 15, 0.9);
console.log(output);